### DISCUSSION TITLE
Initial condition for shock tube simulation

### DISCUSSION THREAD

**Discussion by vachan-potluri** [su2code/SU2#1952] [Show and tell]
Hi,

I had realised that SU2 currently doesn't support discontinuous initial conditions. They are commonly encountered in shock tube simulations. I just implemented this on my fork: https://github.com/su2code/SU2/compare/master...vachan-potluri:SU2_fork:shocktube.

This was inspired from here: https://github.com/vino-123/SU2_v.50_shock_tube/commit/956a3594d3aa4653126fd20ffed7bfadf926cf5b.

It would be great to have this feature shipped with SU2.

**Comment by bigfooted** [comment]
There have been some very recent developments to improve the python wrapper, including creating more advanced initial and boundary conditions, see e.g. #1938 

Maybe you can have a look and see if this works for you. Would be great to see more usage of the python wrapper.

**Reply by vachan-potluri** [reply]
I didn't know that. Thanks for informing! I will check this out.

I however feel this aspect can be better documented in the tutorials.

**Reply by pcarruscag** [reply]
This part of the python wrapper is work in progress, it is not part of a release yet.
But yes we need contributors :)
